### ISSUE TITLE
Car_type as an vehicle attribute

### ISSUE THREAD

**Issue by bergenklem** [openmobilityfoundation/mobility-data-specification#869]
Hi!

Referring to previous discussion in issue [#640](https://github.com/openmobilityfoundation/mobility-data-specification/issues/640) regarding adopting possibility to distinguish between car types. We would very much like to see this as a feature of MDS (extra field in vehicle_attributes?):

> Additionally, there is a new car_type field which could be reference or adopted as well ([source doc](https://docs.google.com/document/d/16NKnf10SjmBBVwUlKrc7oeuxzeWqS_dQvztlT2F4dH8/edit#heading=h.menb22ix2i9m), [source PR](https://github.com/MobilityData/gbfs/pull/255/files?short_path=d9428cb#diff-d9428cb894c125b871daa1393c69bb49c15b37127ff6e410f89d9d39015e60b9)), eg:
> 

> - microcar (smallest size of car, engine smaller than 700cc, e.g Smart Fortwo)
> - subcompact (sedan with interior volume index no larger than 99 cu ft, e.g. Kia Rio)
> - compact (sedan with interior volume index no larger than 109 cu ft, e.g. Hyundai Elantra)
> - mid_size (sedan with interior volume index no larger than 119 u ft, e.g. Toyota Camry)
> - full_size (sedan with interior volume index larger than 120 cu ft, e.g. Tesla Model S)
> - station_wagon (large cargo area with rear hinged tailgate, e.g. Subaru Outback)
> - suv (similar to station-wagon with a light truck frame, e.g. Honda Pilot)
> - minivan (also known as multi-purpose vehicle or people mover, e.g. Dodge Grand Caravan)
> - pickup_truck (enclosed cab with open cargo area and tailgate, e.g. Ford F-150)
> - other

We can see several use cases for such a field:

1 Insight into how different types of cars are being used in carsharing services (in combination with duration, mileage and so forth).

2 Tenders and allocation of reserved spots - give possibility to award companies with wanted mix of car types in their fleet (in our case; we prefer companies with wide range of car types in their fleet when we assign reserved parking lots in neighbourhood areas - not just vans). This has also been used as an family friendliness-indicator in some cities in EU.

3 Control - some of our reserved parking lots have requirements on type of cars they can place there (e.g. in streets to narrow to allow parking for big/wide cars)

Haven't really made up my mind if there's a need for MDS to detail and name the category types (Input from car sharing providers should in that case be considered), or if it should be left out for local goverments/organizations to decide (by leaving it as an open field). Open for discussion!

**Comment by schnuerle** [comment]
Great use cases!

A note to followup on this from a GBFS perspective. It looks like they did not end up implementing this field. Maybe for the same reasons we didn't (it may be hard to define types clearly): 

https://github.com/MobilityData/gbfs/blob/master/gbfs.md#vehicle_typesjson

There is a `form_factor` field, which has 'car' but not much else from the list.

They do have a free-form `name` field which can be anything. They examples on the page include "Example Cargo Bike", "Four-door Sedan", "Example Basic Bike", "Example E-scooter V2". 

Also like MDS there are make and model (but no year) fields.

**Comment by schnuerle** [comment]
If we want this to make it to the MDS 2.1 release, we are going to need someone to make a PR in the next month or two so that it can be discussed and reviewed by the steering committee and working group.

**Comment by saharald** [comment]
Finally, we suggest to keep as few car types as few as possible: city-car, family-car, mpv, van.

**Comment by saharald** [comment]
MPV is probably the most tricky. It could be a 7-seater, an SUV a minibus.

**Comment by geneorama** [comment]
I don't think we should try to define or include this field. If someone in the US wants to know about the type of vehicle I start with the VIN and NHTSA VIN decoder: https://vpic.nhtsa.dot.gov/api/. This has limitations, but other parts of the world are more complicated. 

VIN should work for MDS users because it's a [required](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/modes/car-share.md#vehicle-attributes) field in the MDS Vehicle attributes table. The NHTSA's Vehicle Product Information Catalog (vPIC) drives the API: https://nhtsa.gov/vin and the entire database and decoding tables are available. 

In the US manufacturers are required to file with NHTSA under the authority of the National Traffic and Motor Vehicle Safety Act (see **[49 CFR Part 565](https://www.ecfr.gov/current/title-49/subtitle-B/chapter-V/part-565)** for VIN requirements and **[49 CFR Part 566](https://www.ecfr.gov/current/title-49/subtitle-B/chapter-V/part-566)** for manufacturer requirements). However the fields are not always standardized and some fields are more marketing terms than technically specified. 

This was useful for looking at the relevant fields definitions, but I think it's really a combination of fields that get to vehicle definition we discussed. A combination of Type and Body Type seem like a good fit (Note that some fields like Trim are not even specified, and Body Type is probably BodyClass in the API.):
https://www.ecfr.gov/current/title-49/subtitle-B/chapter-V/part-565/subpart-B/section-565.12

There are regional complications because although VIN is defined by an international standard (ISO 3779), the manufacturers define meanings of the field values. Those definitions are needed to decode the VIN, and they are not part of the international standard. 

So in Europe (since that was the beginning of the discussion) it seems the VIN lookup system is decentralized and there is no organization like NHTSA. 

I think it might be worth providing some guidance and the reasoning why the MDS isn't including it.

**Comment by schnuerle** [comment]
From our [WG meeting on Aug 28 2025](https://github.com/openmobilityfoundation/mobility-data-specification/wiki/Web-conference-notes,-2025.08.28-(MDS-Working-Group)):

- Would like to keep it more generic than the GBFS descriptions, because the specific make/model in examples may not apply to all countries, and change over time (eg Subaru Outback wagon to SUV)
- Might be better to rely on make/model/year lookup
- Maintaining a classification would be hard over time. Even general terms like Microcar can change based on country.
- Classify by number of seats? Add a 'number_of_seats' field, or rider_capacity like GBFS
- Align to [MDS vehicle types](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/data-types.md#vehicle-types)?
- Cubic centimeters of interior space? CCs of engine? Weight? Max length/width/height? Top speed? Number of wheels?
   - Add some of these to the [vehicles object](https://github.com/openmobilityfoundation/mobility-data-specification/blob/main/data-types.md#vehicles)? cargo_volume_capacity, cargo_load_capacity, door_count, max_permitted_speed, rated_power and others from [GBFS](https://github.com/MobilityData/gbfs/blob/v2.3/gbfs.md#vehicle_typesjson)
- What about customization? Make/model/year might not be all that's needed - could have huge tires, or large fenders, etc.
- But this is usually car share companies will have more standard form factors and not heavy mods
- Another place to have custom attributes
- Use of form factor in tenders is important use case
- Proposers might need to make an interim standard
- Could create well-known values like in CDS, but could add your own as needed, allows jurisdiction to specify some things that work for them.
- Similar problem with bicycles: no pedals, throttle, ride on only, assist, cargo

**Comment by schnuerle** [comment]
We have a vehicle run id in Events in CDS 1.1 now. Would a run ID be useful here for you?
https://github.com/openmobilityfoundation/curb-data-specification/tree/main/events#curb-event

"Run identifier from an external runs table containing information about make, model, and/or year, of the vehicle that performed the event."

But we already have make, model, year, and color as [fields](https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/modes/car-share.md#vehicle-attributes) so maybe this is redundant. 

IF GBFS would have included car types or car form factors, then this would be easier, but [they did not](https://github.com/MobilityData/gbfs/blob/master/gbfs.md#vehicle_typesjson).

**Comment by schnuerle** [comment]
Moving to Future release per a conversation with @saharald to allow more time to finalize ideas.
